internal/worker/ibcli-exec.go

```diff
@@ -0,0 +1,75 @@
+package worker
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+
+	"github.com/osbuild/blueprint/pkg/blueprint"
+)
+
+type ImageBuilderArgs struct {
+	Distro    string
+	Arch      string
+	ImageType string
+	Blueprint *blueprint.Blueprint
+
+	// TODO: extend to include all options
+	//  - ostree
+	//  - registration
+	//  - bootc
+	//  - seed
+	//  - repositories
+}
+
+func RunImageBuilderManifest(args ImageBuilderArgs, extraEnv []string, errorWriter io.Writer) ([]byte, error) {
```

Comment:
Why to buffer stdout but not stderr? I mean, I would definitely use `io.Writer` for both. Sanne is working on log streaming and not buffering seems like a better fit in here, also more consistent and on top of that allocates less memory possibly.